### PR TITLE
Refactor DeclareSygusVarCommand and SynthFunCommand to use the API.

### PR DESCRIPTION
**Pull request by abdoo8080**
This PR is part of migrating commands. `DeclareSygusVarCommand` and `SynthFunCommand` now call public API function instead of their corresponding SmtEngine functions. Those two commands don't fully initialize the solver anymore. Some operations in `SygusInterpol::solveInterpolation`, which creates an interpolation sub-solver, depend on the solver being fully initialized and were affected by this change. Those operations are now executed by the main solver instead of the sub-solver, which is not fully initialized by the time they are needed.

### CODE REVIEW

File: src/theory/quantifiers/sygus/sygus_interpol.cpp [hunk 1]
```diff
@@ -319,6 +319,13 @@ bool SygusInterpol::solveInterpolation(const std::string& name,
                                        const TypeNode& itpGType,
                                        Node& interpol)
 {
+  collectSymbols(axioms, conj);
```
Comment (ajreynol) [inline src/theory/quantifiers/sygus/sygus_interpol.cpp:322]: Can you give a quick comment on what this is done here (based on what you said in the PR)?

**Review by ajreynol — CHANGES_REQUESTED** [review]
One minor change, otherwise looks good to me.

File: src/theory/quantifiers/sygus/sygus_interpol.cpp (hunk 1 above)
Comment (abdoo8080) [reply src/theory/quantifiers/sygus/sygus_interpol.cpp:327]: OK. I wrote an explanation for reordering the instructions. Do you think it's clear?

**Review by ajreynol — APPROVED** [review]
LGTM

**Pull request merged**